src/bootloader.rs

```diff
@@ -0,0 +1,216 @@
+    )?;
+
+    let relocated_trace = runner
+        .relocated_trace
+        .as_ref()
+        .ok_or(Error::Trace(TraceError::TraceNotRelocated))?;
+    let trace_path = tmp_dir.path().join("bootloader_trace.json");
+    let trace_file = std::fs::File::create(trace_path.clone())?;
+    let mut trace_writer =
+        FileWriter::new(io::BufWriter::with_capacity(3 * 1024 * 1024, trace_file));
+    write_encoded_trace(relocated_trace, &mut trace_writer)?;
+    trace_writer.flush()?;
+
+    let memory_path = tmp_dir.path().join("bootloader_memory.json");
+    let memory_file = std::fs::File::create(memory_path.clone())?;
+    let mut memory_writer =
+        FileWriter::new(io::BufWriter::with_capacity(5 * 1024 * 1024, memory_file));
+    write_encoded_memory(&runner.relocated_memory, &mut memory_writer)?;
+    memory_writer.flush()?;
+
+    let air_public_input_path = tmp_dir.path().join("bootloader_air_public_input.json");
+    let air_public_input_json = runner.get_air_public_input()?.serialize_json()?;
+    let air_public_input_str = get_formatted_air_public_input(&air_public_input_json)?;
+    std::fs::write(air_public_input_path.clone(), air_public_input_str)?;
+
+    let air_private_input_path = tmp_dir.path().join("bootloader_air_private_input.json");
+    let trace_absolute_path = trace_path
+        .as_path()
+        .canonicalize()
+        .unwrap_or(trace_path.clone())
+        .to_string_lossy()
+        .to_string();
+    let memory_absolute_path = memory_path
+        .as_path()
+        .canonicalize()
+        .unwrap_or(memory_path.clone())
+        .to_string_lossy()
+        .to_string();
+    let air_private_input = runner
+        .get_air_private_input()
+        .to_serializable(trace_absolute_path, memory_absolute_path)
+        .serialize_json()
+        .map_err(PublicInputError::Serde)?;
+    std::fs::write(air_private_input_path.clone(), air_private_input)?;
+
+    let mut output_buffer = "Bootloader program output:\n".to_string();
+    runner.vm.write_output(&mut output_buffer)?;
+    print!("{output_buffer}");
+
+    Ok(CairoBootloaderRunResult {
+        air_public_input: air_public_input_path,
+        air_private_input: air_private_input_path,
+        memory_file: memory_path,
+        trace_file: trace_path,
+    })
+}
+
+fn cairo_run_bootloader_in_proof_mode(
+    bootloader_program: &Program,
+    tasks: Vec<TaskSpec>,
+    layout: LayoutName,
+    fact_topologies_path: PathBuf,
+) -> Result<CairoRunner, CairoRunError> {
+    let mut hint_processor = BootloaderHintProcessor::new();
+
+    let cairo_run_config = CairoRunConfig {
+        entrypoint: "main",
+        trace_enabled: true,
+        relocate_mem: true,
+        layout,
+        proof_mode: true,
+        secure_run: None,
+        disable_trace_padding: false,
+        allow_missing_builtins: None,
+    };
+
+    let program_hash = Felt252::from_dec_str(
+        "382450030162484995497251732956824096484321811411123989415157331925872358847",
```

Comment:
Move into a constant, do we have a public reference for this constant we can include in a comment?